manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -14,7 +14,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-func DefaultPostgresqlSecret(cr *miqv1alpha1.ManageIQ) *corev1.Secret {
+func ManagePostgresqlSecret(cr *miqv1alpha1.ManageIQ, client client.Client, scheme *runtime.Scheme) (*corev1.Secret, controllerutil.MutateFn) {
+	secretKey := types.NamespacedName{Namespace: cr.ObjectMeta.Namespace, Name: cr.Spec.DatabaseSecret}
+	secret := &corev1.Secret{}
+	secretErr := client.Get(context.TODO(), secretKey, secret)
+	if secretErr != nil {
+		secret = defaultPostgresqlSecret(cr)
+	}
+
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, secret, scheme); err != nil {
```

Comment:
@Fryguy Any thoughts on this?  Setting the controller reference will cause this secret to be deleted if the ManageIQ CR is deleted.  Installers could create this on their own (if they're using an external postgres) or let the operator create it (use the internal postgres).